pages/advanced/api-safe-transaction-service/guides/api.md

```diff
@@ -0,0 +1,625 @@
+  const signatureOwnerA = await protocolKitOwnerA.signHash(safeTxHash)
+
+  console.log('Tx signed by Owner A: %s', signatureOwnerA.data)
+
+  // Initialize the Api Kit
+  const apiKit = new SafeApiKit({
+    chainId: 11155111n
+  })
+
+  // Create a Tx in Tx service with Signature of Owner A
+  const senderAddress = await ownerA.getAddress()
+
+  await apiKit.proposeTransaction({
+    safeAddress: config.SAFE_ADDRESS,
+    safeTransactionData: safeTransaction.data,
+    safeTxHash,
+    senderAddress,
+    senderSignature: signatureOwnerA.data
+  })
+
+  console.log('Tx proposed: %s', safeTxHash)
+
+  // Instantiate an EthAdapter for Owner B
+  const ownerB = new ethers.Wallet(config.OWNER_B_PRIVATE_KEY, ethProvider)
+  const ethAdapterOwnerB = new EthersAdapter({
+    ethers,
+    signerOrProvider: ownerB
+  })
+
+  // Initialize the Protocol Kit for Owner B
+  const protocolKitOwnerB = await Safe.create({
+    ethAdapter: ethAdapterOwnerB,
+    safeAddress: config.SAFE_ADDRESS
+  })
+
+  // Sign Tx by Owner B
+  const signatureOwnerB = await protocolKitOwnerB.signHash(safeTxHash)
+
+  console.log('Tx signed by Owner B: %s', signatureOwnerB.data)
+
+  // Add signature of owner B in Tx service
+  await apiKit.confirmTransaction(safeTxHash, signatureOwnerB.data)
+
+  console.log('Tx Owner B signature added in Tx service.')
+}
+```
+
+It is also possible to group multiple transactions into a single transaction and execute them in batch mode. To do so, just define the transactions as follows:
+
+```typescript
+// Create Tx
+const safeTransactionData: MetaTransactionData = {
+  to: config.TO,
+  value: config.VALUE,
+  data: '0x',
+  operation: OperationType.Call
+}
+
+const safeTransactionBData: MetaTransactionData = {
+  to: config.TO_B,
+  value: config.VALUE,
+  data: '0x',
+  operation: OperationType.Call
+}
+
+const safeTransaction = await protocolKitOwnerA.createTransaction({
+  transactions: [safeTransactionData, safeTransactionBData]
+})
+```
+
+#### Using safe-eth-py
+
+```python
+from gnosis.eth import EthereumClient, EthereumNetwork
+from gnosis.safe.api.transaction_service_api import TransactionServiceApi
+from gnosis.safe import Safe
+from hexbytes import HexBytes
+
+def create_transaction_and_sign_by_owners():
+
+    # Instantiate a Safe
+    ethereum_client = EthereumClient(config.get("RPC_URL"))
+    safe = Safe(config.get("SAFE_ADDRESS"), ethereum_client)
+
+    # Create a Tx
+    safe_tx = safe.build_multisig_tx(config.get("TO"), config.get("VALUE"), HexBytes(""))
+
+    # Sign Tx by Owner A
+    safe_tx.sign(config.get("OWNER_A_PRIVATE_KEY"))
+
+    # Instantiate a Tx Service Api
+    transaction_service_api = TransactionServiceApi(
+        network=EthereumNetwork.SEPOLIA,
+        ethereum_client=ethereum_client
+    )
+
+    # Send Tx to Tx Service
+    transaction_service_api.post_transaction(safe_tx)
+
+    # Sign Tx by Owner B
+    owner_b_signature = safe_tx.sign(config.get("OWNER_B_PRIVATE_KEY"))
```

Comment:
I think that because is an example of `offchain` signature interacting with the transaction service, maybe would be better GET the  signed transaction by the owner A from the transaction service to later sign the transaction by the other owner.

Reply:
You are right. I updated the example.